### ISSUE TITLE
Interceptors Persist

### ISSUE THREAD

**Issue by 1toddlewis** [axios/axios#4345]
#### Section/Content To Improve
https://github.com/axios/axios#interceptors

#### Suggested Improvement
Interceptors with the same name appear to create a container object (possibly an array), which surprisingly persists between page calls, and all elements are evaluated. Let's say that on the first time calling Axios, I set a response interceptor:

```
const myInterceptor = axios.interceptors.response.use(async (response) => {
  console.log(`response status: ${response.status}`);
  return response;
});
```

On my first pass through the page, that interceptor will write to the console once.
On my second pass through the page, the console will be written to twice.
On the third pass through the page, the console will be written to three times.

And so on. However, ejecting the interceptor appears to clear the problem entirely:

```
const myInterceptor = axios.interceptors.response.use(async (response) => {
  console.log(`response status: ${response.status}\);
  axios.interceptors.response.eject(myInterceptor);
  return response;
});
```

Behaves as expected, writing to the console only once per pass through.

#### Relevant File(s)
[e.g. README.md]

**Comment by arthurfiorette** [comment]
Hmm, can you provide an example on [runkit](https://npm.runkit.com/axios)? Appearently you are calling the `axios.interceptors.response.use` function every page draw / re-draw.

**Comment by 1toddlewis** [comment]
First time putting something up on runkit (hey, I'm new even if I'm old): [Using Axios to fetch API data](https://runkit.com/1toddlewis/61ccd9ebebba5000080ea9b1).

**Comment by 1toddlewis** [comment]
Also worth noting: I was surprised to see that transformResponse appears to be called before the response is passed to the interceptor. I thought the interceptor would have taken precedence.

**Comment by jasonsaayman** [comment]
Hi 👋 

Please could you retry with the latest pre-release version and open a new issue should this error still be relevant?

Thanks
